File: app/controllers/pages_controller.rb
# frozen_string_literal: true

class PagesController < ApplicationController
  before_action :authenticate

  def index; end

  private

  def authenticate
    @user_type = []
    user = User.find_by_id(session[:current_user_id])
    if user
      @name = user.name
      @user_type.push 'Admin' if user.is_admin
      @user_type.push 'Staff' if user.is_staff
      @user_type.push 'Student' if user.is_student
    end
    @logged_out = !user
  end
end

File: app(react)/src/controllers/pages_controller.js
class PagesController extends ApplicationController {
    constructor(...args) {
        super(...args);
    }
    index() {
        this.render('pages/index');
    }
    about() {
        this.render('pages/about');
    }
    contact() {
        this.render('pages/contact');
    }
    privacy() {
        this.render('pages/privacy');
    }
    terms() {
        this.render('pages/terms');
    }
    faq() {
        this.render('pages/faq');
    }
    private;

    authenticate() {
    this._user_type = [];
    var user = User.find_by_id(session.current_user_id);

    if (user) {
        this._name = user.name;
        if (user.is_admin) this._user_type.push("Admin");
        if (user.is_staff) this._user_type.push("Staff");
        if (user.is_student) this._user_type.push("Student")
    };

    this._logged_out = !user
    }
}